xgbse/_base.py

```diff
@@ -61,7 +61,7 @@ def get_neighbors(
             index_leaves = self.bst.predict(
                 index_matrix,
                 pred_leaf=True,
-                iteration_range=(0, self.bst.best_iteration),
+                iteration_range=(0, self.bst.best_iteration + 1),
```

Comment:
I guess it should be `self.bst.num_trees()`, and then you wouldnt need to add `+ 1`.
The `best_iteration` can be the first one, 0, even with an ensemble of 50 trees, for example.